src/queue.ts

```diff
@@ -0,0 +1,18 @@
+export class Queue<T> {
```

Comment:
Why don't we just export them as default since we won't be exporting anything else from this file ? 
It's much cleaner.

Reply:
Makes sense, I think we can create a sub-folder `structures` to hold these files then `operations` to hold operations being performed like `traversals`